.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+name: Continuous integration
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  # This matrix job runs the test suite against multiple Ruby and Rails versions
+  test_matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
+        ruby: [2.7, '3.0']
+        # Test against multiple Rails versions
+        gemfile: [rails_6, rails_7]
+    runs-on: ubuntu-latest
+    env:
+      DB_DATABASE: signonotron2_test
+      DB_USER: root
+      DB_PASSWORD: root
+      MYSQL_ALLOW_EMPTY_PASSWORD: true
+      BUNDLE_GEMFILE: gemfiles/${{ matrix.gemfile }}.gemfile
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7 # for signon
+          bundler-cache: true
+          if: ${{ matrix.ruby != 2.7 }}
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Set up MySQL
+        run: |
+          sudo /etc/init.d/mysql start
+          mysql -e "CREATE DATABASE ${{ env.DB_DATABASE }}; CREATE USER 'signonotron2'@'localhost' IDENTIFIED BY 'signonotron2'; GRANT ALL ON ${{ env.DB_DATABASE }}.* to 'signonotron2'@'localhost';" -u${{ env.DB_USER }} -p${{ env.DB_PASSWORD }}
+      - name: Start Signon
+        run: |
+          export PATH=/opt/hostedtoolcache/Ruby/2.7.6/x64/bin:$PATH
+          git clean -fxde /tmp
+          env BUNDLE_GEMFILE=Gemfile TEST_DATABASE_URL=mysql2://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:3306/${{ env.DB_DATABASE }} ./start_signon.sh
+      - name: Run tests
+        run: |
+          bundle install
+          bundle exec rake
+      - if: always()
```

Comment:
The `if` definitely threw me off, but [looks like the right thing to use](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions)! 😅